--- a/demux/demux_lavf.c
+++ b/demux/demux_lavf.c
@@ -1292,7 +1292,10 @@
     dp->dts = mp_pts_from_av(pkt->dts, &st->time_base);
     dp->duration = pkt->duration * av_q2d(st->time_base);
     dp->pos = pkt->pos;
-    dp->keyframe = pkt->flags & AV_PKT_FLAG_KEY;
+    // DFPWM has no keyframe syntax. FFmpeg's encoder omits AV_PKT_FLAG_KEY,
+    // which otherwise makes mpv discard the entire stream when selecting it.
+    dp->keyframe = (pkt->flags & AV_PKT_FLAG_KEY) ||
+                   st->codecpar->codec_id == AV_CODEC_ID_DFPWM;
     av_packet_unref(pkt);
 
     if (priv->format_hack.clear_filepos)
